src/App.js

```diff
@@ -6,27 +6,53 @@ import Card from './components/UI/Card';
 class App extends React.Component {
   constructor(props) {
     super(props);
+
     this.state = {
+      checkboxUnderHeader: false,
       data: {
         inputValue: '',
         textareaValue: '',
       }
     }
   }
 
+  onCheckboxUnderHeaderChange = () => {
+    this.setState((state) => ({
+      checkboxUnderHeader: !state.checkboxUnderHeader,
+    }));    
+    console.log(this.state.checkboxUnderHeader)
+  }
+
   updateData = (tempData) => {
     this.setState({
       data: { ...tempData },
     });
+    
```

Comment:
This code will work every time ANY card is updated. Which will lead all the cards are updated. The point is to update each card separately from others.